pkg/nodeconfig/init.go

```diff
@@ -17,6 +19,8 @@ type cache struct {
 	workerIgnitionEndPoint string
 	// apiServerEndpoint is the address which clients can interact with the API server through
 	apiServerEndpoint string
+	// credentials holds a certificate and token needed to interact with the API server
+	credentials *cluster.Authentication
```

Comment:
I don't think this fits in here. The other members of this are immutable. The cert can be rotated.